src/index.ts

```diff
@@ -7,6 +7,14 @@ export type BraintreeTransactionOrRefund =
   | BraintreeTransaction
   | BraintreeRefund;
 
+export type BraintreePaymentContextOrError =
+  | BraintreePaymentContext
+  | HandlerError;
+
+// A custom error to return from a Custom Actions handler
+export interface HandlerError {
+  message: string;
```

Comment:
Any thoughts on adding `field` here to as optional? I guess you would need a list...

Reply:
I _think_ what you are suggesting is adding something like `field` so that we can model payload validation errors, correct? If so, I think that would be another type and maybe we should consider something like a `BasicError`?

```typescript
type HandlerError = ValidationError | BasicError

interface BasicError {
  message: string
}

interface ValidationError {
  message?: string
  fields: FieldValidationError[]
}

interface FieldValidationError {
  name: string;
  message: string;
}

Reply:
@cmonty do you think this is a modeling we can "grow into" or something that we should do now?

Reply:
You can add it later.